core/src/App.html

```diff
@@ -304,22 +301,34 @@
           }
         });
         window.addEventListener('message', async e => {
+          const allIframes = Iframe.getAllIframes(node, this.get().modalIframe);
+          const isOriginWindow = (iframe) => iframe && (e.source === iframe.contentWindow || e.source === window);
+          const iframe = allIframes.find(isOriginWindow);
```

Comment:
In case of a message coming from the current window we get here any/random iframe. I guess this case (message coming not from an iframe but from the core - which means current window) we need a better handling. This makes the whole thing not simpler though.. We could add an fake iframe to the list of frames in the beginning with .contentWindow set to current window and .luigi.viewUrl to the current window's url...